presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -198,7 +199,23 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Set<Symbol>> context)
             PlanNode left = context.rewrite(node.getLeft(), leftInputs);
             PlanNode right = context.rewrite(node.getRight(), rightInputs);
 
-            return new JoinNode(node.getId(), node.getType(), left, right, node.getCriteria(), node.getFilter(), node.getLeftHashSymbol(), node.getRightHashSymbol());
+            List<Symbol> outputSymbols;
+            if (node.getCriteria().isEmpty() && !node.getFilter().isPresent() && node.getType() == INNER) {
+                // do not prune nested joins output since it is not supported
```

Comment:
done. This is more implementation specific (e.g: nested loop doesn't support filter), but I agree one condition method is better.